src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1086,16 +1086,22 @@ private bool CheckPropertyValueKind(SyntaxNode node, BoundExpression expr, BindV
                 {
                     if (setMethod.IsInitOnly)
                     {
+                        bool hasInitError = false;
                         if (!isAllowedInitOnlySet(receiver))
                         {
                             Error(diagnostics, ErrorCode.ERR_AssignmentInitOnly, node, propertySymbol);
-                            return false;
+                            hasInitError = true;
                         }
 
                         if (setMethod.DeclaringCompilation != this.Compilation)
                         {
                             // an error would have already been reported on declaring an init-only setter
-                            CheckFeatureAvailability(node, MessageID.IDS_FeatureInitOnlySetters, diagnostics);
+                            hasInitError |= !CheckFeatureAvailability(node, MessageID.IDS_FeatureInitOnlySetters, diagnostics);
```

Comment:
> hasInitError |= !CheckFeatureAvailability(node, MessageID.IDS_FeatureInitOnlySetters, diagnostics); [](start = 27, length = 100)

This change doesn't feel right to me. Targeting language version shouldn't change how we bind things, it should only affect feature availability diagnostics. Also, I don't think we would want the binding behavior to be different based on where the property is defined. Bottom line, the only difference is acceptable is presence or lack of feature availability diagnostics, nothing else.